Sergey_tasks/JS/Bank-project/Bank.js

```diff
@@ -0,0 +1,275 @@
+const movementsRowWithdrawal = document.querySelector(".movements__row-withdrawal");
+const movementsTypeDeposit = document.querySelector(".movements__type-deposit");
+const movementsTypeWithdrawal = document.querySelector(".movements__type-withdrawal");
+const movementsValueDeposit = document.querySelector(".movements__value-deposit");
+const movementsValueWithdrawal = document.querySelector(".movements__value-withdrawal");
+//---------------------------------------------------------------------------
+const btnLogin = document.querySelector('.login__btn');
+const btnTransfer = document.querySelector('.form__btn--transfer');
+const btnLoan = document.querySelector('.form__btn--loan');
+const btnClose = document.querySelector('.form__btn--close');
+const btnSort = document.querySelector('.btn--sort');
+//---------------------------------------------------------------------------
+const inputLoginUsername = document.querySelector('.login__input--user');
+const inputLoginPin = document.querySelector('.login__input--pin');
+const inputTransferTo = document.querySelector('.form__input--to');
+const inputTransferAmount = document.querySelector('.form__input--amount');
+const inputLoanAmount = document.querySelector('.form__input--loan-amount');
+const inputCloseUserName = document.querySelector('.form__input--user');
+const inputClosePin = document.querySelector('.form__input--pin');
+
+//--Переменные ---------------------------------------------------------------
+const today = new Date();
+let closeActiveUser;
+
+//--Запись в данных массива accounts в localStorage если они не записаны
+function init(){
+   if(JSON.parse(localStorage.getItem("balance")))
+      return init = null;
+   else localStorage.setItem("balance", JSON.stringify(accounts))
+}
+init();
+
+//--Проверка Login и пароль и если TRUE то вывод на экран данных активного клиента
+btnLogin.addEventListener("click", () =>{
+
+   labelDate.textContent = `${today.getDate()}.0${today.getMonth() + 1}.${today.getFullYear()}`;
+   //--Считываем Login и пароль из Input ------------------
+   const clientLogin = inputLoginUsername.value;
+   const clientPin = parseInt(inputLoginPin.value);
+   //-------------------------------------------------------
+   const clientVerify = JSON.parse(localStorage.getItem("balance")).find(elem => elem.login === clientLogin && elem.pin === clientPin);
+   if(clientVerify){
+      //--Записываем  в localStorage Login активного клиента -------------
+      const activeClient = clientVerify.login;
+      localStorage.setItem("activeClient", activeClient);
+   }
+   initClients();
+   closeActiveUser = setInterval(timer, 1000);
+});
+
+//--Перевод средств (Transfer)-------------------------------------------
+btnTransfer.addEventListener("click", () => {
+   const transfer = (JSON.parse(localStorage.getItem("balance")) || []);
+   //--Переводим средства и записываем в Local Storage -----------------
+   transfer.map(elem => {
```

Comment:
this works incorrectly, you shouldn't update the initial balance object. The map method should return a new array.